test-clients/src/test/java/TestingFactoryTest.java

```diff
@@ -0,0 +1,136 @@
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.props.JutilPropertySource;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hedera.services.bdd.suites.contract.openzeppelin.ERC20ContractInteractions;
+import com.hedera.services.bdd.suites.contract.openzeppelin.ERC721ContractInteractions;
+import com.hedera.services.bdd.suites.contract.precompile.AssociatePrecompileSuite;
+import com.hedera.services.bdd.suites.contract.precompile.ContractBurnHTSSuite;
+import com.hedera.services.bdd.suites.contract.precompile.ContractHTSSuite;
+import com.hedera.services.bdd.suites.contract.precompile.ContractKeysHTSSuite;
+import com.hedera.services.bdd.suites.contract.precompile.ContractMintHTSSuite;
+import com.hedera.services.bdd.suites.contract.precompile.CryptoTransferHTSSuite;
+import com.hedera.services.bdd.suites.contract.precompile.DelegatePrecompileSuite;
+import com.hedera.services.bdd.suites.contract.precompile.DissociatePrecompileSuite;
+import com.hedera.services.bdd.suites.contract.precompile.DynamicGasCostSuite;
+import com.hedera.services.bdd.suites.contract.precompile.MixedHTSPrecompileTestsSuite;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.hedera.services.bdd.suites.HapiApiSuite.FinalOutcome.SUITE_PASSED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+
+
+class TestingFactoryTest  {
+	private static final Logger log = LogManager.getLogger(TestingFactoryTest.class);
+
+	@BeforeAll
+	static void beforeAll() {
+		final String systemProperty = System.getProperty("DEFAULT_PORT");
+		final var defaultPort = systemProperty != null ? systemProperty : "50211";
+		final var defaultProperties = JutilPropertySource.getDefaultInstance();
+		HapiApiSpec.runInCiMode(
+				defaultPort + ":50212",
+				defaultProperties.get("default.payer"),
+				defaultProperties.get("default.node").split("\\.")[2],
+				defaultProperties.get("tls"),
+				defaultProperties.get("txn.proto.structure"),
+				defaultProperties.get("node.selector"),
+				Collections.emptyMap()
+		);
+	}
+
+	@Tag("contract.precompile")
+	@TestFactory
+	Collection<DynamicTest> contractPrecompileTests() {
+		return List.of(
+				dynamicTest("Asosciate Precompile Suite",
+						() -> {
+							final var associatePrecompileSuite = new AssociatePrecompileSuite();
+							assertEquals(SUITE_PASSED, getOutcomeSyncOrAsync(associatePrecompileSuite));
+						}),
```

Comment:
This is a good first start. I think we could go further and inject each spec as a separate dynamic test instead of the suite as a whole. The suites could be mapped to JUnit's `DynamicContainer`. Ideally, I'd like to see something like this (pseudocode):

```java
	@Tag("contract.precompile")
	@TestFactory
	Collection<DynamicTest> contractPrecompileTests() {
		return List.of(
				hapiApiSuite(AssociatePrecompileSuite::new),
				hapiApiSuite(ContractMintHTSSuite::new),
                ...
        );
    }

    private DynamicContainer hapiApiSuite(Supplier<HapiApiSuite> suiteSupplier) {
        HapiApiSuite suite = suiteSupplier.get();
        var tests = suite.getSpecsInSuite().stream().map(s -> dynamicTest(s.getName(), s::run)
        return dynamicContainer(suite.getClass().getSimpleName(), tests);
    }
```

Or not sure if you can leverage the `HapiSpecRegistry`.